internal/server/iigointernal/utilities.go

```diff
@@ -0,0 +1,100 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/pkg/errors"
+)
+
+// PickUpRulesByVariable returns a list of rule_id's which are affected by certain variables
+func PickUpRulesByVariable(variableName string, ruleStore map[string]rules.RuleMatrix) ([]string, error) {
+	var Rules []string
+	if _, ok := rules.VariableMap[variableName]; ok {
+		for k, v := range ruleStore {
+			_, err := searchForStringInArray(variableName, v.RequiredVariables)
+			if err != nil {
+				Rules = append(Rules, k)
+			}
+		}
+		return Rules, nil
+	} else {
+		return []string{}, errors.Errorf("Variable name '%v' was not found in the variable cache", variableName)
+	}
+}
+
+func searchForStringInArray(val string, array []string) (int, error) {
+	for i, v := range array {
+		if v == val {
+			return i, nil
+		}
+	}
+	return 0, errors.Errorf("Not found")
+}
+
+type DataPacket struct {
```

Comment:
Agreed.
```
type DataPacketContentType = int

const (
    DataPacketInt DataPacketContentType = iota
    DataPacketString
)

type DataPacket struct {
     T DataPacketContentType 
     Int *int
     Str *string
     Bool *bool
}
```

Pointers or no pointers up to you--pointers have extra safety of `nil` check, and you can (but less encouraged) even get rid of the enum tag and just try to unwrap everything until you have a non-nil ptr.

Reply:
I've removed the DataPacket type since it was just a duplicate of baseclient.Communications.
We've added the content type fields and modified the instantiation. (Coming in next push)